internal/testutil/status/condition.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
+
+import (
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func SortConditionsByType(c []metav1.Condition) []metav1.Condition {
+	cp := make([]metav1.Condition, len(c))
+	copy(cp, c)
+	sort.SliceStable(cp, func(i, j int) bool {
+		return cp[i].Type < cp[j].Type
+	})
+	return cp
+}
+
+func ReplaceConditions(conditions []metav1.Condition, sadConditions []metav1.Condition) []metav1.Condition {
```

Comment:
naming nit: you don't have to replace with "sad" conditions, right? Maybe something like `replacementConditions`?

Reply:
updated in next push.